x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/cluster/routing/allocation/DataTierAllocationDecider.java

```diff
@@ -90,8 +91,7 @@ public void validate(String value, Map<Setting<?>, Object> settings, boolean exi
                             "] tier preference may be used for partial searchable snapshots (got: [" + value + "])");
                     }
                 } else {
-                    String[] split = value.split(",");
-                    if (Arrays.stream(split).anyMatch(DATA_FROZEN::equals)) {
+                    if (FROZEN_TIER_PATTERN.matcher(value).find()) {
```

Comment:
I am not really sure about this bit, might run faster, might not?

Perhaps we can add a test validating that just doing `value.indexOf(DATA_FROZEN)` is safe, i.e., that there are no other tiers starting with `DATA_FROZEN`?

Reply:
Jup see d62ad36 that should be much faster and is more fun to read as well :)